src/engine/source/router/src/policyManager.cpp

```diff
@@ -0,0 +1,117 @@
+#include <router/policyManager.hpp>
+
+#include <logging/logging.hpp>
+#include <name.hpp>
+
+namespace router
+{
+
+std::optional<base::Error> PolicyManager::addPolicy(const std::string& name)
+{
+    // Validate the runtime policy name
+    base::Name envName;
+    try
+    {
+        envName = base::Name {name};
+    }
+    catch (const std::exception& e)
+    {
+        return base::Error {fmt::format("Invalid policy name: '{}'", e.what())};
+    }
+
+    if (envName.parts().size() != 3)
+    {
+        return base::Error {fmt::format("Invalid policy name: '{}', the expected "
+                                        "format is: \"policy/<env-name>/<version>\"",
```

Comment:
Messages should also be updated to 'policy'.
```suggestion
                                        "format is: \"policy/<policy-name>/<version>\"",
```

Reply:
done